roles/sno_installer/tasks/22_rhcos_image_cache.yml

```diff
@@ -91,13 +108,27 @@
       environment:
         DBUS_SESSION_BUS_ADDRESS: "{{ ansible_env.DBUS_SESSION_BUS_ADDRESS | default('unix:path=/run/user/' + ansible_effective_user_id | string + '/bus') }}"
 
+    - name: "Get status of the image cache service"
+      ansible.builtin.systemd:
+        name: container-cache.service
+        state: started
+        scope: user
+      environment:
+        DBUS_SESSION_BUS_ADDRESS: "{{ ansible_env.DBUS_SESSION_BUS_ADDRESS | default('unix:path=/run/user/' + ansible_effective_user_id | string + '/bus') }}"
+      check_mode: true
+      register: _si_cache_service_status
+
     - name: "Start container-cache.service"
       ansible.builtin.systemd:
         name: container-cache.service
```

Comment:
stupid question, just put the state to "started" is not doing the same thing ?

Reply:
yes and no.
If you set it as "started" and the service is already running, the service will continue running with the current settings, regardless of whether we changed the configuration of the container (for instance, the path to the mounted cache directory) or the settings in the service unit. Thus the need to restart the service.